### ISSUE TITLE
Missing propagation of ealiest/latest dates in corner case

### ISSUE THREAD

**Issue by jcoupey** [VROOM-Project/vroom#339]
TL;DR the triangle inequality strikes back

After a local search move implying a job addition in a route, we typically update the new job's earliest (resp. latest) dates, then propagate computing of earliest dates forward (resp. latest dates backward) in the route. For `v1.6.0`, this used to happen in `TWRoute::fwd_update_earliest_from` and `TWRoute::bwd_update_latest_from`. Those functions had checks to stop propagation [here](https://github.com/VROOM-Project/vroom/blob/0ec7279e95bbc3d3474ba06ded3a8aa4f8931dab/src/structures/vroom/tw_route.cpp#L76-L78) and [here](https://github.com/VROOM-Project/vroom/blob/0ec7279e95bbc3d3474ba06ded3a8aa4f8931dab/src/structures/vroom/tw_route.cpp#L98-L100). The rationale is that when a new earliest date is not delayed past the previously known earliest date at some point, there is no point in checking further the rest of the route.

Now this check should actually have been written with `==` rather than `<=`, as it was in similar functions [here](https://github.com/VROOM-Project/vroom/blob/0ec7279e95bbc3d3474ba06ded3a8aa4f8931dab/src/structures/vroom/tw_route.cpp#L120-L123) and [here](https://github.com/VROOM-Project/vroom/blob/0ec7279e95bbc3d3474ba06ded3a8aa4f8931dab/src/structures/vroom/tw_route.cpp#L143-L146). Actually the only situation where this matters is when after adding a job, the new earliest date afterwards is **strictly lower**. This only occurs when adding a job reduces the overall travel time (triangle inequality violation) **and** the service time at the new job does not even cover the difference. In that case we stop propagation, so we keep earliest/latest dates that are too pessimistic.

This affects `v1.6.0` but the good news are:

1. Nothing breaks, the worst that can happen is that because of the pessimistic values for earliest/latest dates, some local search moves are discarded as invalid while they would be considered valid using the right dates.

2. This is fixed in `master` and for the next release. During the work on breaks at #303 the above functions have been refactored and merged with the right checks [here](https://github.com/VROOM-Project/vroom/blob/bff7493d36cc483878ab6ea6669d722856746fe2/src/structures/vroom/tw_route.cpp#L177) and [here](https://github.com/VROOM-Project/vroom/blob/bff7493d36cc483878ab6ea6669d722856746fe2/src/structures/vroom/tw_route.cpp#L288).

**Comment by jcoupey** [comment]
I'll leave this ticket open for reference until we release `v1.7.0`.

**Comment by jcoupey** [comment]
Fixed in `v1.7.0`.
